core/server/common/src/main/java/alluxio/master/journal/MergeJournalContext.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.journal;
+
+import alluxio.exception.status.UnavailableException;
+import alluxio.proto.journal.Journal.JournalEntry;
+
+import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.UnaryOperator;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Context for merging journal entries together for a wrapped journal context.
+ */
+@NotThreadSafe
+public final class MergeJournalContext implements JournalContext {
+  private static final Logger LOG = LoggerFactory.getLogger(MergeJournalContext.class);
+  private final JournalContext mJournalContext;
+  private final UnaryOperator<List<JournalEntry>> mMergeOperator;
+  private final List<JournalEntry> mJournalEntries;
```

Comment:
Can we add a warning message when we have too many entries in this list? Just so that we can be aware of potential memory pressure in case it is used for large list of entries accidentally.

Reply:
done